sdcm/cluster.py

```diff
@@ -1494,6 +1494,59 @@ def start_scylla(node, queue):
         time_elapsed = time.time() - start_time
         self.log.debug('Update DB binary duration -> %s s', int(time_elapsed))
 
+    def _update_db_packages(self, new_scylla_bin, node_list):
+        self.log.debug('User requested to update DB packages...')
+
+        seed_nodes = self.get_seed_nodes()
+        non_seed_nodes = [n for n in self.nodes if not n.is_seed]
+
+        def update_scylla_packages(node, queue):
+            node.log.info('Updating DB binary')
+            node.remoter.send_files(new_scylla_bin, '/tmp/scylla', verbose=True)
+            node.remoter.run('sudo yum update -y')
+            node.remoter.run('sudo yum install python34-PyYAML -y')
+            # replace the packages
+            node.remoter.run('yum list installed | grep scylla')
+            # update *developmen* packages
+            node.remoter.run('sudo rpm -UvhR /tmp/scylla/*development* | true')
+            # and all the rest
+            node.remoter.run('sudo rpm -URvh --replacefiles /tmp/scylla/* | true')
+            node.remoter.run('yum list installed | grep scylla')
+            queue.put(node)
+            queue.task_done()
+
+        def stop_scylla(node, queue):
+            node.wait_db_up()
+            node.remoter.run('sudo systemctl stop scylla-server.service')
+            node.remoter.run('sudo systemctl stop scylla-jmx.service')
+            node.wait_db_down()
+            queue.put(node)
+            queue.task_done()
+
+        def start_scylla(node, queue):
+            node.wait_db_down()
+            node.remoter.run('sudo systemctl start scylla-server.service')
+            node.remoter.run('sudo systemctl start scylla-jmx.service')
+            node.wait_db_up()
+            queue.put(node)
+            queue.task_done()
+
+        start_time = time.time()
+
+        # First, stop *all* non seed nodes
```

Comment:
This is a problem when using the update_db_binary or packages during longevity tests.
This issue is that during longevity, we have decommission and add_node right after it.
In this case, the add node, triggers *update*, but instead of stopping only the new node, it stops the whole cluster and kills the longevity test.

In the past I patched it locally using:
+        if len(node_list) == 1:
+            # Stop only new nodes
+            self.run_func_parallel(func=stop_scylla, node_list=node_list)
+            # Then, update bin only on requested new nodes
+            self.run_func_parallel(func=update_scylla_bin, node_list=node_list)
+            # Start new nodes
+            self.run_func_parallel(func=start_scylla, node_list=node_list)
+        else:
+            # First, stop *all* non seed nodes
+            self.run_func_parallel(func=stop_scylla, node_list=non_seed_nodes)
+            # Second, stop *all* seed nodes
+            self.run_func_parallel(func=stop_scylla, node_list=seed_nodes)
+            # Then, update bin only on all nodes
+            self.run_func_parallel(func=update_scylla_bin, node_list=node_list)
+            # Start all seed nodes
+            self.run_func_parallel(func=start_scylla, node_list=seed_nodes)
+            # Start all non seed nodes
+            self.run_func_parallel(func=start_scylla, node_list=non_seed_nodes)

However, this won't work if we will have in the future cases that we add more than 1 node to an existing cluster and want to update the package.